docs/src/main/sphinx/connector/oracle.rst

```diff
@@ -388,6 +388,25 @@ The connector supports pushdown for a number of operations:
 * :ref:`limit-pushdown`
 * :ref:`topn-pushdown`
 
+.. _oracle-predicate-pushdown:
+
+Predicate pushdown support
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector does not support pushdown of any predicates on columns that use
+the Oracle database ``CLOB``, ``NCLOB``, ``BLOB``, or ``RAW(n)`` types. This
+also applies to Trino data types that :ref:`map to these Oracle database types
+<oracle-type-mapping>`.
+
+In the following example, the predicate is not pushed down for either query
+since ``name`` is a column of type ``VARCHAR``, which maps to ``NCLOB`` in
+Oracle:
+
+.. code-block:: sql
+
+    SELECT * FROM nation WHERE name > 'CANADA';
+    SELECT * FROM nation WHERE name = 'CANADA';
+
```

Comment:
This explanation is a bit misleading.

Oracle is a bit complex. Consult https://trino.io/docs/current/connector/oracle.html#character-mapping, you'll notice that for VARCHAR(n < 2000) we map to Oracle's VARCHAR2 (for which pushdown is possible).
While for unbounded VARCHAR or n > 2000 we map to Oracle's NCLOB for which pushdown isn't possible.

So this example is not sufficient.

IMO this might be more revealing:

```suggestion
In the following example, the predicate is pushed down for both queries
since ``name`` is a column of type ``VARCHAR(25)``, which maps to ``VARCHAR2(25)`` in
Oracle:

.. code-block:: sql

    SHOW CREATE TABLE nation;
    --             Create Table
    ----------------------------------------
    -- CREATE TABLE oracle.trino_test.nation (
    --    name varchar(25)
    -- )
    -- (1 row)
    SELECT * FROM nation WHERE name > 'CANADA';
    SELECT * FROM nation WHERE name = 'CANADA';

In the following example, the predicate is not pushed down for either query
since ``name`` is a column of type ``VARCHAR``, which maps to ``NCLOB`` in
Oracle:

.. code-block:: sql

    SHOW CREATE TABLE nation;
    --             Create Table
    ----------------------------------------
    -- CREATE TABLE oracle.trino_test.nation (
    --    name varchar
    -- )
    -- (1 row)
    SELECT * FROM nation WHERE name > 'CANADA';
    SELECT * FROM nation WHERE name = 'CANADA';
```

Reply:
Makes sense, thanks for clarifying & providing the example.